src/cdk/platform/platform.ts

```diff
@@ -59,4 +60,8 @@ export class Platform {
   // this and just place the Safari keyword in the userAgent. To be more safe about Safari every
   // Safari browser should also use Webkit as its layout engine.
   SAFARI: boolean = this.isBrowser && /safari/i.test(navigator.userAgent) && this.WEBKIT;
+
+  constructor(@Inject(PLATFORM_ID) private _platformId: Object) {
```

Comment:
This is a breaking change, because it changes the constructor signature.

Reply:
Totally agree, but I still think it's a change that eventually needs to go in. I'm going through the rest of the tests to update the signature

Reply:
This still isn't the proper way to handle it though. Instead it should be marked as an optional parameter with a deletion target for 7.0.0 and it should fall back to the old way of checking.

Reply:
Agreed, will change